test/blackbox-tests/test-cases/pkg/extra-source-overlap-with-source.t

```diff
@@ -2,20 +2,27 @@ Test for packages with an extra-source file with the same name as a
 file in the package's source.
 
   $ . ./helpers.sh
-  $ make_lockdir
-
-  $ cat > dune.lock/foo.pkg <<EOF
-  > (version 1)
-  > (source
-  >  (copy $PWD/foo-source))
-  > (extra_sources
-  >  (foo.txt
-  >   (fetch
-  >    (url file://$PWD/foo.txt))))
-  > EOF
 
+Create a fake package archive
```

Comment:
This change and others is what I'm referring to. We should be able to test how `extra_sources` works without invoking the solver